web/js/modules/layers/util.js

```diff
@@ -1405,6 +1349,92 @@ export function adjustStartDates(layers) {
   return Object.values(layers).forEach(applyDateAdjustment);
 }
 
+/**
+ * For active, multi-interval layers with on going coverage,
+ * date ranges are modified and added for layer config
+ *
+ * @method adjustActiveDateRanges
+ * @param  {Array} layers array
+ * @param  {Object} appNow - pageLoadTime date
+ * @returns {Array} array of layers
+ */
+export function adjustActiveDateRanges(layers, appNow) {
+  // active layers that can't be future predicted reliably due to sparse coverage
+  const ignoredLayers = {
+    AMSRU2_Snow_Water_Equivalent_5Day: true,
```

Comment:
Two thoughts on this bit:

1.  Hard-coding layer ids into the code feels hacky.  When I checked, it looked like only 4 layers were being adjusted by this function.  With so few being adjusted, maybe it makes sense to add a property flag to the layer config itself so we are confident that only the layers we want adjusted are modified rather than having to rely on date checking logic alone?  A property like: `predictFutureDateRanges: true` or something.
2. Why did this layer qualify as having sparse coverage but the OSCAR layers didn't? They seemed to have lots of weird small gaps between ranges.

Reply:
This function now extends the endDate of the last dateRange object of any multiday layer. So say you build in 11/2020, and have a 4-day layer in 5/2021, the dateRange will reset at 01/01/2021 and allow the proper 4-day block of 2021 coverage to continue instead of carrying over the 2020 coverage, which may be incorrect depending on when that 4-day block ended in 2020.

I'll remove the ignoredLayers and that can provide an opportunity to help find any issues with those layers in the GC.